### ISSUE TITLE
tde-base is unbuildable with glibc 2.28

### ISSUE THREAD

**Issue by CPAllianceP** [michael-manley/Trinity_ArchLinux_PKGBUILD#31]
https://github.com/tavianator/bfs/issues/35

```Scanning dependencies of target kdesktop_lock
[ 70%] Building CXX object kdesktop/lock/CMakeFiles/kdesktop_lock.dir/lockprocess.cc.o
In file included from /build/tde-tdebase/src/tdebase/kdesktop/lock/lockprocess.cc:83:0:
/usr/include/linux/stat.h:56:8: error: redefinition of ‘struct statx_timestamp’
 struct statx_timestamp {
        ^~~~~~~~~~~~~~~
In file included from /usr/include/sys/stat.h:446:0,
                 from /build/tde-tdebase/src/tdebase/kdesktop/lock/lockprocess.cc:73:
/usr/include/bits/statx.h:25:8: note: previous definition of ‘struct statx_timestamp’
 struct statx_timestamp
        ^~~~~~~~~~~~~~~
In file included from /build/tde-tdebase/src/tdebase/kdesktop/lock/lockprocess.cc:83:0:
/usr/include/linux/stat.h:99:8: error: redefinition of ‘struct statx’
 struct statx {
        ^~~~~
In file included from /usr/include/sys/stat.h:446:0,
                 from /build/tde-tdebase/src/tdebase/kdesktop/lock/lockprocess.cc:73:
/usr/include/bits/statx.h:36:8: note: previous definition of ‘struct statx’
 struct statx
        ^~~~~
make[2]: *** [kdesktop/lock/CMakeFiles/kdesktop_lock.dir/build.make:96: kdesktop/lock/CMakeFiles/kdesktop_lock.dir/lockprocess.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:24951: kdesktop/lock/CMakeFiles/kdesktop_lock.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...

**Comment by SlavekB** [comment]
It seems that include linux/stat.h is not needed and can be removed.
Please try the following patch:

```
Index: b/kdesktop/lock/lockprocess.cc
===================================================================
--- a/kdesktop/lock/lockprocess.cc
+++ b/kdesktop/lock/lockprocess.cc
@@ -80,9 +80,6 @@
 
 #include <kcrash.h>
 
-#ifdef __linux__
-#include <linux/stat.h>
-#endif
 #include <pthread.h>
 
 #include <X11/Xlib.h>
```

**Comment by CPAllianceP** [comment]
That worked perfectly, thanks. Would you mind updating to R14.0.5?

**Comment by SlavekB** [comment]
Thanks for testing the patch.

Because R14.0.5 is already final and released, the patch will be integrated for the upcoming R14.0.6.

**Comment by SlavekB** [comment]
For your information, the patch is already included - see commit [de4afaaf00](https://mirror.git.trinitydesktop.org/gitea/TDE/tdebase/commit/de4afaaf00)
